### PR TITLE
Minimize permissions

### PR DESCRIPTION
**Pull request by YuvalFireFly**

### CODE REVIEW

File: iam_policy_part6.tf [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+          "route53resolver:GetFirewallDomainList",
+          "route53resolver:GetFirewallRuleGroup",
+          "route53resolver:GetFirewallRuleGroupAssociation",
+          "route53resolver:GetResolverEndpoint",
+          "route53resolver:GetResolverQueryLogConfig",
+          "route53resolver:GetResolverQueryLogConfigAssociation",
+          "route53resolver:GetResolverRule",
+          "route53resolver:GetResolverRuleAssociation",
+          "route53resolver:List*",
+          "s3-outposts:List*",
```
Comment (sefi-infralight) [inline iam_policy_part6.tf:32]: s3-outposts:ListAccessPoints is sensitive

File: iam_policy_part1.tf [hunk 2]
```diff
@@ -4,207 +4,155 @@ resource "aws_iam_policy" "firefly_readonly_policy_part1" {
...
+          "codestar:Describe*",
+          "codestar:List*",
+          "cognito-identity:Describe*",
+          "cognito-identity:GetIdentityPoolRoles",
+          "cognito-identity:GetPrincipalTagAttributeMap",
+          "cognito-identity:List*",
+          "cognito-idp:Describe*",
+          "cognito-idp:List*",
+          "cognito-sync:Describe*",
+          "cognito-sync:List*"
```
Comment (sefi-infralight) [inline iam_policy_part1.tf:151]: cognito-sync:ListIdentityPoolUsage might be sensitive

File: iam_policy_part6.tf (hunk 1 above)
Comment (YuvalFireFly) [reply iam_policy_part6.tf:32]: We are not trying to remove sensitive data, we are trying to minimize the permissions needed by flywheel...

File: iam_policy_part2.tf [hunk 3]
```diff
@@ -4,186 +4,152 @@ resource "aws_iam_policy" "firefly_readonly_policy_part2" {
...
+          "dynamodb:DescribeExport",
+          "dynamodb:DescribeGlobalTable",
+          "dynamodb:DescribeGlobalTableSettings",
+          "dynamodb:DescribeKinesisStreamingDestination",
+          "dynamodb:DescribeLimits",
+          "dynamodb:DescribeStream",
+          "dynamodb:DescribeTable",
+          "dynamodb:DescribeTableReplicaAutoScaling",
+          "dynamodb:DescribeTimeToLive",
+          "dynamodb:GetItem",
```
Comment (sefi-infralight) [inline iam_policy_part2.tf:80]: dynamodb:GetItem - big no no!!!

File: iam_policy_part3.tf [hunk 4]
```diff
@@ -4,189 +4,154 @@ resource "aws_iam_policy" "firefly_readonly_policy_part3" {
...
+        "Action" : [
+          "elastic-inference:DescribeAcceleratorOfferings",
+          "elastic-inference:DescribeAcceleratorTypes",
+          "elastic-inference:DescribeAccelerators",
+          "elastic-inference:ListTagsForResource",
+          "elasticache:Describe*",
+          "elasticache:List*",
+          "elasticbeanstalk:Describe*",
+          "elasticbeanstalk:List*",
+          "elasticfilesystem:Describe*",
```
Comment (sefi-infralight) [inline iam_policy_part3.tf:19]: efs DescribeAccessPoints - sensitive

File: iam_policy_part4.tf [hunk 5]
```diff
@@ -4,154 +4,160 @@ resource "aws_iam_policy" "firefly_readonly_policy_part4" {
...
+          "iam:GetPolicy",
+          "iam:GetPolicyVersion",
+          "iam:GetRole",
+          "iam:GetRolePolicy",
+          "iam:GetSAMLProvider",
+          "iam:GetSSHPublicKey",
+          "iam:GetServerCertificate",
+          "iam:GetUser",
+          "iam:GetUserPolicy",
+          "iam:ListAccessKeys",
```
Comment (sefi-infralight) [inline iam_policy_part4.tf:26]: iam:ListAccessKeys - bad

**Review by galco1 — APPROVED** [review]

**Pull request merged**